### DISCUSSION TITLE
Plan for angular 12 upgrade

### DISCUSSION THREAD

**Discussion by sushan05** [vmware/clarity#6004] [General]
What are the plans for angular 12 upgrade and enabling Ivy ?

**Comment by ph55** [comment]
Yes, can we get issue to track please?

**Reply by bbogdanov** [reply]
Hi @sushan05 and @ph55,

We are actively working on bringing support for Angular v12 and it will become available as soon as we are ready.

**Reply by kumart-netapp** [reply]
Thanks.
Any tentative date available?

**Reply by bbogdanov** [reply]
@kumart-netapp The initiative that holds the Angular 12 support a little bit bigger than just providing support for v12. We can't say a specific date or release. Our goal is to release the initiative as soon as possible.

**Reply by sushan05** [reply]
Hi @bbogdanov 
we see that `next` branch in on Angular 12 and its peer dependencies is on Angular 11. 
what's the status of updating peer dependencies.

**Reply by bbogdanov** [reply]
@sushan05 Hi!

As I mentioned in the previous comments the upgrade to Angular 12 is a little bit bigger intiative than just upgrading `@clr/angular` to Angular 12. 

https://github.com/vmware/clarity/blob/angular/projects/angular/package.json The upgrade to Angular 12 will arrive in NPM this week.

**Reply by dpsarrou** [reply]
Hi folks, any updates on this?

**Reply by gnomeontherun** [reply]
Yes it is available as `v12.0.0-next.0` as a preview https://www.npmjs.com/package/@clr/angular/v/12.0.0-next.0 try it out and let us know if there are any issues. Hopefully we can release final soon.

**Reply by mathisloge** [reply]
`ng add @clr/angular@12.0.0-next.0` have added `@clr/angular@6.0.0` for me and the installation failed

**Reply by bbogdanov** [reply]
@mathisloge I'll take a look at `ng add` and write back when I have something to share.

 In the meantime we advices you to try it out in the old fashion way with `npm install @clr/angular@12.0.0-next.0` or `yarn add @clr/angular@12.0.0-next.0`

**Reply by bbogdanov** [reply]
https://github.com/vmware/clarity/pull/6159 I suppose it will come in the next release and it solves everything related to `@clr/angular@12{anything}`

**Reply by bbogdanov** [reply]
@mathisloge The `ng add` command is fixed and should be working just fine. 

We also released v12 as stable so you could give it a try with `ng add @clr/angular` on a project with Angular 12. 

Let us know if you face any issues.

**Reply by mathisloge** [reply]
nice. thanks for notifying me. Will report if i face any issues.

**Reply by bbogdanov** [reply]
@ph55 @sushan05 @kumart-netapp @dpsarrou Angular 12 support was released as stable yesterday. https://www.npmjs.com/package/@clr/angular/v/12.0.0

Let us know if you face any issue.

**Reply by kumar-tadepalli** [reply]
@bbogdanov 
What is the version number clarity is going to follow?
Is Clarity aligning with angular version for @clr/* package?
How do we co-relate @cds/core version which is v5.4.3 now?

Basically in my project is not using @cds/core, it is @clr/* based once. But we have to include in package.json to make compiler happy and avoid below error:

Error: ./node_modules/@clr/angular/__ivy_ngcc__/fesm2015/clr-angular.js
Module not found: Error: Can't resolve '@cds/core/icon' in 'C:\Users\admin\repo\WS\ui\node_modules\@clr\angular\__ivy_ngcc__\fesm2015'
resolve '@cds/core/icon' in 'C:\Users\admin\repo\WS\ui\node_modules\@clr\angular\__ivy_ngcc__\fesm2015'
  Parsed request is a module
  using description file: C:\Users\admin\repo\WS\ui\node_modules\@clr\angular\package.json (relative path: ./__ivy_ngcc__/fesm2015)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      looking for modules in C:/Users/admin/repo/WS/ui/src
        using description file: C:\Users\admin\repo\WS\ui\package.json (relative path: ./src)

**Reply by bbogdanov** [reply]
@shravansofts 

We are writing a blog post where we are going in details about what we are doing with `@clr/angular`. @gnomeontherun could give you more details when it will be available.

Basically `@clr/*` is aligning with angular version, so you guessed that correctly. We are going to support every Angular version until we recreate all our components in Clarity Core.

Clarity Core is going to follow its own versioning and we are going to release `v6` in near future.

On the other hand, After `@clr/angular@>=5.0.0` `@clr/angular`  depends on `@cds/core` for the icons so `@cds/core` is a required package to have if you want to use Clarity Angular v5 and above. `@clr/icons` is deprecated and it won't be supported anymore. You can learn more about `@clr/icons` deprecation [here](https://medium.com/claritydesignsystem/goodbye-clarity-icons-hello-clarity-core-adb07de9739e)
